cmd/subctl/diagnose.go

```diff
@@ -215,54 +208,49 @@ func addDiagnoseFWConfigFlags(command *cobra.Command) {
 		"produce verbose output while validating the firewall")
 }
 
-func diagnoseAll(clusterInfo *cluster.Info, status reporter.Interface) bool {
-	success := diagnose.K8sVersion(clusterInfo.ClientProducer.ForKubernetes(), status)
-
-	fmt.Println()
-
-	if clusterInfo.Submariner == nil {
-		status.Warning(constants.SubmarinerNotInstalled)
-
-		return success
+func failIfNoContext(contextPresent bool, err error) error {
+	if !contextPresent {
+		return errors.New("no context was specified or found")
```

Comment:
It seems this will fail the command but not print a message. 

```suggestion
                 fmt.Fprintln(os.Stderr, "No kube config or context was specified or found.")
		return errors.New("no context was specified or found")
```